activiti-cloud-query-service/activiti-cloud-services-query/activiti-cloud-services-query-repo/src/main/java/org/activiti/cloud/services/query/app/repository/CustomizedTaskRepositoryImpl.java

```diff
@@ -59,5 +60,47 @@ public CustomizedTaskRepositoryImpl() {
 
         return PageableExecutionUtils.getPage(tasks.fetch(), pageable, countQuery::fetchCount);
    }
+    
+   @Override
+    public Iterable<TaskEntity> findAllOverProcessInstance(Predicate predicate) {
+
+        QTaskEntity taskEntity = QTaskEntity.taskEntity;
+
+        JPQLQuery<TaskEntity> from = buildLeftJoin(taskEntity, predicate);
+        JPQLQuery<TaskEntity> tasks = from.select(taskEntity);
+
+        return tasks.fetch();
+    }
+
+    @Override
+    public Page<TaskEntity> findAllOverProcessInstance(Predicate predicate, Pageable pageable) {
+        
+        QTaskEntity taskEntity = QTaskEntity.taskEntity;
+        
+        JPQLQuery<TaskEntity> from = buildLeftJoin(taskEntity, predicate);
+        final JPQLQuery<?> countQuery = from.select(taskEntity.count());
+        JPQLQuery<TaskEntity> tasks = from.select(taskEntity);
+
+        return PageableExecutionUtils.getPage(tasks.fetch(), pageable, countQuery::fetchCount);
+        
+    }
+
+    @Override
+    public boolean existsOverProcessInstance(Predicate predicate) {
+        QTaskEntity taskEntity = QTaskEntity.taskEntity;
+        JPQLQuery<TaskEntity> from = buildLeftJoin(taskEntity, predicate);
```

Comment:
what does it happen in case the predicate is null ?

Reply:
The predicate should never be null.  There is a Assert in `buildLeftJoin`